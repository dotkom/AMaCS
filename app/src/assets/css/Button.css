.button {
  background: #0060a3;
  border: 3px solid #0060a3;
  box-sizing: border-box;
  box-shadow: 0px 3px 4px rgba(0, 0, 0, 0.25);
  border-radius: 2px;
  width: 100%;
  max-width: 24rem;
  padding: 0.5rem 1.5rem;
  color: white;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.3rem;
  transition: 0.2s;
  display: block;
  user-select: none;
  cursor: default;
}

.buttonContent {
  display: flex;
  align-items: center;
  justify-content: center;
}

.button:hover {
  background: #f2f2f2;
  color: #0060a3;
}

.button:active, button.active {
  background: #ffffff;
  color: #0060a3;
}

.button:disabled {
  background: #f2f2f2;
  border: 3px solid transparent;
  color: #c0c0c0;
}

.iconLeft {
  width: 2rem;
  margin-right: 0.5rem;
}

.iconRight {
  width: 2rem;
  margin-left: 0.5rem;
}

a .button:hover {
   
}