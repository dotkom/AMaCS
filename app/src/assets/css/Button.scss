@import 'variables';

.button {
  background: $blue;
  border: 3px solid $blue;
  box-sizing: border-box;
  box-shadow: 0px 3px 4px rgba(0, 0, 0, 0.25);
  border-radius: 2px;
  width: 100%;
  max-width: 24rem;
  padding: 0.5rem 1.5rem;
  color: white;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.3rem;
  transition: 0.2s;
  display: block;
  user-select: none;
  cursor: default;
}

.buttonContent {
  display: flex;
  align-items: center;
  justify-content: center;
}

.button:hover {
  background: $light-gray;
  color: $blue;
}

.button:active, button.active {
  background: white;
  color: $blue;
}

.button:disabled {
  background: $light-gray;
  border: 3px solid transparent;
  color: #c0c0c0;
}

.iconLeft {
  width: 2rem;
  margin-right: 0.5rem;
}

.iconRight {
  width: 2rem;
  margin-left: 0.5rem;
}

a .button:hover {
   
}