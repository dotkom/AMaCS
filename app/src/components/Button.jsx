import React from "react";

import _s from 'assets/css/Button.css';


function Button({ children, onClick, text, iconLeft, iconRight, disabled }) {
  return(
    <button className={_s.button} onClick={onClick} disabled={disabled}>
      <div className={_s.buttonContent}>
      { iconLeft && <img className={_s.iconLeft} src={iconLeft} /> }
      { text }
      { iconRight && <img className={_s.iconRight} src={iconRight} /> }
      </div>
    </button>
  )
}

export default Button;
