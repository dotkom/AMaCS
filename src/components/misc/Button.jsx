import React, { useState, useLayoutEffect, useCallback } from 'react';

import _s from 'assets/css/Button.module.scss';

const Button = ({ text, iconLeft, iconRight, hoverIconLeft, hoverIconRight, onMouseEnter, onMouseLeave, ...props }) => {
  const [hover, setHover] = useState(false);

  const preloadIcons = useCallback(() => {
    if (hoverIconLeft) {
      new Image().src = hoverIconLeft;
    }
    if (hoverIconRight) {
      new Image().src = hoverIconRight;
    }
  }, [hoverIconLeft, hoverIconRight]);

  const HandleMouseEnter = (...a) => {
    setHover(true);
    if (onMouseEnter) {
      onMouseEnter(...a);
    }
  };

  const handleMouseLeave = (...a) => {
    setHover(false);
    if (onMouseLeave) {
      onMouseLeave(...a);
    }
  };

  useLayoutEffect(() => {
    preloadIcons();
  }, [preloadIcons]);

  const icLeft = hover ? hoverIconLeft || iconLeft : iconLeft || hoverIconLeft;
  const icRight = hover ? hoverIconRight || iconRight : iconRight || hoverIconRight;

  return (
    <button {...props} className={_s.button} onMouseEnter={HandleMouseEnter} onMouseLeave={handleMouseLeave}>
      <div className={_s.buttonContent}>
        {icLeft && <img className={_s.iconLeft} src={icLeft} alt="Ikon" />}
        {text}
        {icRight && <img className={_s.iconRight} src={icRight} alt="Ikon" />}
      </div>
    </button>
  );
};

export default Button;
