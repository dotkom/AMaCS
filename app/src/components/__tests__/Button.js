import React from 'react';
import { shallow } from 'enzyme';

import Button from '../Button';


describe('button', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Button />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with text', () => {
    const wrapper = shallow(
      <Button text="Button with text" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with icon', () => {
    const wrapper = shallow(
      <Button text="Button with icon" iconLeft="icon.png" />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with right icon', () => {
    const wrapper = shallow(
      <Button text="Button with icon right" iconRight="icon2.png" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
