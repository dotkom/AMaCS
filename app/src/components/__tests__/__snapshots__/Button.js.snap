// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly 1`] = `
<button
  className="button"
>
  <div
    className="buttonContent"
  />
</button>
`;

exports[`button renders correctly with icon 1`] = `
<button
  className="button"
>
  <div
    className="buttonContent"
  >
    <img
      className="iconLeft"
      src="icon.png"
    />
    Button with icon
  </div>
</button>
`;

exports[`button renders correctly with right icon 1`] = `
<button
  className="button"
>
  <div
    className="buttonContent"
  >
    Button with icon right
    <img
      className="iconRight"
      src="icon2.png"
    />
  </div>
</button>
`;

exports[`button renders correctly with text 1`] = `
<button
  className="button"
>
  <div
    className="buttonContent"
  >
    Button with text
  </div>
</button>
`;
